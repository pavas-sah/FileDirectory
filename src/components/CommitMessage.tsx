import React from 'react';

export interface CommitMessageProps {
  message: string;
}

export const CommitMessage: React.SFC<CommitMessageProps> = (props) => {
  return <>{props.message}</>;
};
export default CommitMessage;
