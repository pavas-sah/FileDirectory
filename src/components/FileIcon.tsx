import React from 'react';

export interface FileIconProps {
  url: string;
  alt?: string;
}

export const FileIcon: React.SFC<FileIconProps> = (props) => {
  return <img src={props.url} alt={props.alt} />;
};
export default FileIcon;
