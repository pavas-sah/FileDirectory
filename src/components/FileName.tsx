import React from 'react';

export interface FileNameProps {
  fileName: string;
}

export const FileName = (props: FileNameProps) => {
  return <>{props.fileName}</>;
};
export default FileName;
