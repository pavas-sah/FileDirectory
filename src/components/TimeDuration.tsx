import React from 'react';

import TimeAgo from 'javascript-time-ago'
import en from 'javascript-time-ago/locale/en'

// Add locale-specific relative date/time formatting rules.
TimeAgo.addLocale(en)

// Create relative date/time formatter.
const timeAgo = new TimeAgo('en-US')

export interface TimeDurationProps {
    dateInString: string
}


const TimeDuration: React.SFC<TimeDurationProps> = (props) => {
    const dt = new Date(props.dateInString);
    const dtInMS = dt.valueOf();
    const currentDateInMS = Date.now().valueOf()
    const absoluteTimeInMS = currentDateInMS - dtInMS;

    return (
    <span data-testid="time">
        {absoluteTimeInMS > 0 ? timeAgo.format(Date.now() - (absoluteTimeInMS), 'time') + " ago" :
         timeAgo.format(Date.now() - (absoluteTimeInMS), 'time') + " after" }
    </span>);

};

export default TimeDuration;