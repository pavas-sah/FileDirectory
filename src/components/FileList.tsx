import React, { Component } from 'react';
import data from '../data/data.json';
import FileListItem from './FileListItem';
import './FileList.css';

export interface FileState {
  id?: number;
  fileIcon: string;
  fileIconAlt: string;
  fileName: string;
  message: string;
  duration: string;
}

export class FileList extends Component {
  fileList = data.map((item: FileState) => {
    return (
      <span className='collection-item' key={item.id}>
        <FileListItem
          fileIcon={require(`../icons/${item.fileIcon}.png`)}
          fileIconAlt={item.fileIconAlt}
          fileName={item.fileName}
          message={item.message}
          duration={item.duration}
        />
      </span>
    );
  });

  render() {
    return <>{this.fileList}</>;
  }
}

export default FileList;
