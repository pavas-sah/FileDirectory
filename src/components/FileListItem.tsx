import React from 'react';
import FileIcon from './FileIcon';
import FileName from './FileName';
import CommitMessage from './CommitMessage';
import TimeDuration from './TimeDuration';
import styled from 'styled-components';
import { FileState } from './FileList';

export const RowDiv = styled.div`
 display: flex;
 flex-flow: row wrap;
 justify-content: space-around;
 width:100%;
 }`;

const FileListItem = (props: FileState) => {
  return (
    <RowDiv>
      <FileIcon url={props.fileIcon} alt={props.fileIconAlt} />
      <FileName fileName={props.fileName} />
      <CommitMessage message={props.message} />
      <TimeDuration dateInString={props.duration} />
    </RowDiv>
  );
};

export default FileListItem;
