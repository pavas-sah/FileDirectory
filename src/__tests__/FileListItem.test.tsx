import React from 'react';
import renderer from 'react-test-renderer';
import { FileState } from '../components/FileList';
import FileListItem from '../components/FileListItem';

const mockDuration = '12-5-2010';
const mockIcon = 'iconLocation';
const mockAlt = 'iconAlt';
const mockFileName = 'file';
const mockMessage = 'message';
const mockId = 1;

const fileListItemProps: FileState = {
  duration: mockDuration,
  fileIcon: mockIcon,
  fileIconAlt: mockAlt,
  fileName: mockFileName,
  message: mockMessage,
  id: mockId,
};

describe('Test File List Item', () => {
  it('renders correctly', () => {
    const fileListComponent = renderer.create(
      <FileListItem {...fileListItemProps} />
    );

    expect(fileListComponent.toJSON()?.type).toBe('div');
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<FileListItem {...fileListItemProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
