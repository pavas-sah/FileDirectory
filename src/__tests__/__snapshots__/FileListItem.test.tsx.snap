// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test File List Item matches snapshot 1`] = `
<div
  className="sc-AxjAm jIgIEX"
>
  <img
    alt="iconAlt"
    src="iconLocation"
  />
  file
  message
  9 years ago
</div>
`;
